safehttp/plugins/csp/csp_test.go

```diff
@@ -0,0 +1,253 @@
+				"frame-ancestors 'self'; report-uri https://example.com/collector",
+			},
+			wantNonce: "KSkpKSkpKSkpKSkpKSkpKSkpKSk=",
+		},
+		{
+			name: "StrictCSP Report Only",
+			interceptor: Interceptor{
+				ReportOnly: []Policy{
+					StrictCSPBuilder{ReportURI: "https://example.com/collector"}.Build(),
+				},
+			},
+			wantReportOnlyPolicy: []string{
+				"object-src 'none'; script-src 'unsafe-inline' https: http: 'nonce-KSkpKSkpKSkpKSkpKSkpKSkpKSk='; base-uri 'none'; report-uri https://example.com/collector",
+			},
+			wantNonce: "KSkpKSkpKSkpKSkpKSkpKSkpKSk=",
+		},
+		{
+			name: "FramingCSP Report Only",
+			interceptor: Interceptor{
+				ReportOnly: []Policy{
+					FramingPolicyBuilder{ReportURI: "https://example.com/collector"}.Build(),
+				},
+			},
+			wantReportOnlyPolicy: []string{"frame-ancestors 'self'; report-uri https://example.com/collector"},
+			wantNonce:            "KSkpKSkpKSkpKSkpKSkpKSkpKSk=",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			rr := safehttptest.NewResponseRecorder()
+			req := safehttptest.NewRequest(safehttp.MethodGet, "/", nil)
+
+			tt.interceptor.Before(rr.ResponseWriter, req)
+
+			h := rr.Header()
+			if diff := cmp.Diff(tt.wantEnforcPolicy, h.Values("Content-Security-Policy"), cmpopts.EquateEmpty()); diff != "" {
+				t.Errorf("h.Values(\"Content-Security-Policy\") mismatch (-want +got):\n%s", diff)
+			}
+
+			if diff := cmp.Diff(tt.wantReportOnlyPolicy, h.Values("Content-Security-Policy-Report-Only"), cmpopts.EquateEmpty()); diff != "" {
+				t.Errorf("h.Values(\"Content-Security-Policy-Report-Only\") mismatch (-want +got):\n%s", diff)
+			}
+
+			v := req.Context().Value(ctxKey{})
+			if v == nil {
+				t.Fatalf("req.Context().Value(ctxKey{}) got: nil want: %q", tt.wantNonce)
+			}
+			if got := v.(string); got != tt.wantNonce {
+				t.Errorf("v.(string) got: %q want: %q", got, tt.wantNonce)
+			}
+		})
+	}
+}
+
+func TestAlreadyClaimed(t *testing.T) {
+	headers := []string{
+		"Content-Security-Policy",
+		"Content-Security-Policy-Report-Only",
+	}
+
+	for _, h := range headers {
+		t.Run(h, func(t *testing.T) {
+			rr := safehttptest.NewResponseRecorder()
+			if _, err := rr.ResponseWriter.Header().Claim(h); err != nil {
+				t.Fatalf("rr.ResponseWriter.Header().Claim(h) got err: %v want: nil", err)
+			}
+			req := safehttptest.NewRequest(safehttp.MethodGet, "/", nil)
+
+			it := Interceptor{}
+			it.Before(rr.ResponseWriter, req)
+
+			if got, want := rr.Status(), int(safehttp.StatusInternalServerError); got != want {
+				t.Errorf("rr.Status() got: %v want: %v", got, want)
+			}
+
+			wantHeaders := map[string][]string{
+				"Content-Type":           {"text/plain; charset=utf-8"},
+				"X-Content-Type-Options": {"nosniff"},
+			}
+			if diff := cmp.Diff(wantHeaders, map[string][]string(rr.Header())); diff != "" {
+				t.Errorf("rr.Header() mismatch (-want +got):\n%s", diff)
+			}
+
```

Comment:
nit: remove empty line

Reply:
I have the if-statements grouped like this:
```go
if check status {
    //...
}

wantHeaders = ...
if check headers {
    //...
}

if check body {
    // ...
}
```
Now, I think that removing the space between the header check and the body check would actually make this less readable. Either I keep both the empty lines or I remove both of them like this:
```go
if check status {
    //...
}
wantHeaders = ...
if check headers {
    //...
}
if check body {
    // ...
}
```
What do you think?

Reply:
Looks good to me :)